src/hangar/diagnostics/integrity.py

```diff
@@ -0,0 +1,212 @@
+import os
+import tempfile
+import warnings
+
+import lmdb
+import numpy as np
+from tqdm import tqdm
+
+from .. import constants as c
+from ..backends import BACKEND_ACCESSOR_MAP, is_local_backend
+from ..context import TxnRegister
+from ..records import commiting, hashmachine, hashs, parsing, queries, heads
+
+
+def _verify_array_integrity(hashenv: lmdb.Environment, repo_path: os.PathLike):
+
+    fs = {}
+    hq = hashs.HashQuery(hashenv)
+    data_kvs = hq.gen_all_hash_keys_raw_array_vals_parsed()
+    narrays = hq.num_arrays()
+    nremote = 0
+    try:
+        for digest, val in tqdm(data_kvs, total=narrays, desc='verifying arrays'):
+            if val.backend not in fs:
+                fs[val.backend] = BACKEND_ACCESSOR_MAP[val.backend](repo_path, None, None)
+                fs[val.backend].open(mode='r')
+            if is_local_backend(val) is False:
+                nremote += 1
+                warnings.warn(
+                    'Can not verify integrity of partially fetched array data references. '
+                    'For complete proof, fetch all remote data locally.', RuntimeWarning)
+                continue
+            o = fs[val.backend].read_data(val)
```

Comment:
Same issue exists here as for `_verify_schema_integrity`, there's no guarrentee that the values would even be parsable, let alone contain the expected field names. 

Additionally, these variable names suck. How was I possibly ok with them wile writing this??